src/nvim/marktree.h

```diff
@@ -107,12 +110,24 @@ static inline bool mt_decor_sign(MTKey key)
   return key.flags & (MT_FLAG_DECOR_SIGNTEXT | MT_FLAG_DECOR_SIGNHL);
 }
 
-static inline uint16_t mt_flags(bool right_gravity, bool no_undo, bool invalidate, bool decor_ext)
+static inline bool mt_scoped(MTKey key)
+{
+  return key.flags & MT_FLAG_SCOPED;
+}
+
+static inline bool mt_scoped_ns_check(MTKey key, win_T *wp)
```

Comment:
I'd call this `mt_scoped_in_win()` and give it the reverse return value (true = positive = visible)